presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayDistinctFunction.java

```diff
@@ -30,11 +32,17 @@
 @Description("Remove duplicate values from the given array")
 public final class ArrayDistinctFunction
 {
-    private ArrayDistinctFunction() {}
+    private final PageBuilder pageBuilder;
+
+    @TypeParameter("E")
+    public ArrayDistinctFunction(@TypeParameter("E") Type elementType)
+    {
+        pageBuilder = new PageBuilder(ImmutableList.of(elementType));
```

Comment:
I have put up a pull request to address the issue @wenleix raised (#10794).

To unblock yourself while you continue to work on this PR (and address comment from @wenleix about `bigintDistinct`), you can add the following code temporarily. You will remove that code before merging, after rebasing on top of #10794.

```
public ArrayDistinctFunction()
{
    this(BIGINT);
}